src/mincostflow.py

```diff
@@ -0,0 +1,118 @@
+        # creates the edges files that contains the head and tail nodes and the weights after them
+        edges.to_csv(filename_map['edges'], sep='\t',index=False, columns = ["Interactor1","Interactor2","Weight"], header=False) #removed columns and now it works columns = [Node1, Node2, Weight]
+
+    @staticmethod
+    def run (sources = None, targets = None, edges= None, output_file = None, flow = None, capacity = None, singularity=False):
+        """
+        Run min cost flow with Docker (or singularity)
+        @param sources:  input sources (required)
+        @param targets: input targets (required)
+        @param edges:  input network file (required)
+        @param output: output file name (required)
+        @param flow: amount of flow going through the graph (optional)
+        @capacity: amount of capacity allowed on each edge (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container
+        """
+
+        # ensures that these parameters are required
+        if not sources or not targets or not edges or not output_file:
+            raise ValueError('Required PathLinker arguments are missing')
+
+        # the data files will be mapped within this directory within the container
+        work_dir = '/mincostflow'
+
+        # the tuple is for mapping the sources, targets, edges, and output 
+        volumes = list()
+
+        # required arguments 
+        bind_path, sources_file = prepare_volume(sources, work_dir) # sources_file = workingdirectory/joiefhowihfj/sources.txt (which is sources)
+        volumes.append(bind_path)
+
+        bind_path, targets_file = prepare_volume(targets, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, edges_file = prepare_volume (edges, work_dir)
+        volumes.append(bind_path)
+
+        out_dir = Path(output_file).parent
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        mapped_out_prefix = mapped_out_dir + '/out'
+       
+        # makes the python command to run within in the container
+        command = ['python',
+                    '/MinCostFlow/minCostFlow.py',
+                    '--sources_file',sources_file,
+                    '--targets_file',targets_file,
+                    '--edges_file', edges_file,
+                    '--output', mapped_out_prefix]
+        
+        #optional arguments (extend the command if available)
+        if flow is not None:
+            command.extend (['--flow', str(flow)])
+        if capacity is not None:
+            command.extend (['--capacity', str(capacity)])
+
+        # choosing to run in docker or sigularity container
+        container_framework = 'singularity' if singularity else 'docker'
+
+        # constructs a docker run call 
+        out = run_container(container_framework,
+                            'ntalluri2/mincostflow',
```

Comment:
We can update this after I push the Docker image to the reedcompbio Docker organization

Reply:
I pushed the image to the reedcompbio Docker organization: https://hub.docker.com/repository/docker/reedcompbio/mincostflow
```
$ docker build -t reedcompbio/mincostflow -f Dockerfile .
$ docker push reedcompbio/mincostflow
```

We can now change this line to use `reedcompbio/mincostflow` and pull the container from DockerHub if the user does not have it locally.  We can also update the GitHub actions workflow to pull the container for DockerHub as a cache when it builds the container as part of the test workflow.